### DISCUSSION TITLE
DIPY Project 6 : DIPY Algorithm Optimizations

### DISCUSSION THREAD

**Discussion by ujjwal-shekhar** [dipy/dipy#2748] [#GSoC/GSoD]
Hello Mentors,

My name is Ujjwal Shekhar, and I am a second-year student pursuing a Bachelor's in Technology in Computer Science and an M.S. by Research in Computational Natural Sciences from the International Institute of Information Technology - Hyderabad. I am interested in contributing to Project 6: DIPY Algorithms Optimization for GSoC 2023.

After reviewing the project description and steps, I found the project intriguing. I enjoy working with algorithms and have a strong background in Python and experience in C/C++, which might be useful. I can make a significant contribution to this project if given the opportunity.

While I have yet to gain experience with CUDA, Openmp, MPI, SIMD, or SIMT, I am excited to use this opportunity to learn and gain more expertise in these areas.

I would like to connect with the mentors (@skoudoro @pjsjongsung) to discuss the project further and understand how I can start contributing. If it's not too late, I would like to know if this project is still available for me.

Thank you, and I look forward to hearing back from you soon.

**Comment by skoudoro** [comment]
Hi @ujjwal-shekhar,

Great !

Do not hesitate to bring your specific question here.

**Reply by ujjwal-shekhar** [reply]
Hi @skoudoro,

Thank you for taking the time to reply. I appreciate it.

I am excited to start contributing. Could you please help me get started by answering a few questions?

- Are there any open issues related to the project that I can work on? Also, could you guide me to any specific part of the codebase that I can improve and open a PR?
- I plan to look for areas of the code that require algorithmic improvement in terms of time complexity or space complexity and then explore parallelisation possibilities. Does this sound like a good approach to you? Do you have any suggestions for which parts of the codebase I should look at?
- What specific requirements would you like me to fulfil before drafting a proposal?

I appreciate your help and guidance. Thank you.

**Reply by skoudoro** [reply]
first, I recommend you to read this blog post: https://levelup.gitconnected.com/my-journey-guide-for-google-summer-of-code-7b03e1007c07

> Are there any open issues related to the project that I can work on? Also, could you guide me to any specific part of the codebase that I can improve and open a PR?

Do not forget, your time is precious, our time is precious (mentor do it for free), GSoC program is to bring value to opensource software. 

You bring a skill and we train you to opensource world.  WE need to see those skills !!   We need to see that you can search in a codebase, analyze a problem and fix it or talk about it. 

> I plan to look for areas of the code that require algorithmic improvement in terms of time complexity or space complexity and then explore parallelisation possibilities. Does this sound like a good approach to you? Do you have any suggestions for which parts of the codebase I should look at?

Great that you plan to do it! looking forward for your valuable conclusion.

> What specific requirements would you like me to fulfil before drafting a proposal?

it is all explain in the blog post link above. 

Good luck ! and if you have any specific tehnical issue, feel free to ask on this thread

**Reply by skoudoro** [reply]
Also, I forgot to add, it is mandatory from Python Software Foundation (PSF) to do a Pull request before the end of the GSOC application period.
